pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -651,6 +669,139 @@ func (s *DiscoveryServer) pushEds(push *model.PushContext, con *XdsConnection,
 	return nil
 }
 
+// Apply filter functions listed in the 'filterFuncs' one after the other
+func (s *DiscoveryServer) applyFilterFuncs(cla *xdsapi.ClusterLoadAssignment, con *XdsConnection) *xdsapi.ClusterLoadAssignment {
+	filtered := cla
+	for _, ff := range filterFuncs {
+		filtered = ff(filtered, con, s.Env)
+	}
+	return filtered
+}
+
+// This is a network filter function to support Split Horizon EDS - filter the endpoints based on the network
+// of the connected sidecar. The filter will filter out all endpoints which are not present within the
+// sidecar network and add a gateway endpoint to remote networks that have endpoints (if gateway exists).
+// Information for the mesh networks is provided as a MeshNetwork config map.
+func networkFilter(cla *xdsapi.ClusterLoadAssignment, con *XdsConnection, env *model.Environment) *xdsapi.ClusterLoadAssignment {
+	// If the sidecar does not specify a network, ignore Split Horizon EDS and return all
+	conNetwork, found := con.modelNode.Metadata["ISTIO_NETWORK"]
```

Comment:
nit: just name this `network`, nothing else uses that name and its most descriptive given the context of the method IMO.